modules/xmpp/JingleSessionPC.js

```diff
@@ -19,1476 +19,1561 @@ import * as JingleSessionState from "./JingleSessionState";
-    this.readSsrcInfo($(offerIq).find(">content"));
-    return remoteSdp;
-};
+    /**
+     * Add the given ssrc lines to the current remote sdp
+     * @param {list} addSsrcInfo a list of SDP line strings that
+     *  should be added to the remote SDP
+     * @returns type {SDP Object} the new remote SDP (after removing the lines
+     *  in removeSsrcInfo
+     */
+    _processRemoteAddSource (addSsrcInfo) {
+        const remoteSdp = new SDP(this.peerconnection.remoteDescription.sdp);
+        addSsrcInfo.forEach(function(lines, idx) {
+            remoteSdp.media[idx] += lines;
+        });
+        remoteSdp.raw = remoteSdp.session + remoteSdp.media.join('');
+
+        return remoteSdp;
+    }
 
-/**
- * Remove the given ssrc lines from the current remote sdp
- * @param {list} removeSsrcInfo a list of SDP line strings that
- *  should be removed from the remote SDP
- * @returns type {SDP Object} the new remote SDP (after removing the lines
- *  in removeSsrcInfo
- */
-JingleSessionPC.prototype._processRemoteRemoveSource = function (removeSsrcInfo) {
-    let remoteSdp = new SDP(this.peerconnection.remoteDescription.sdp);
-    removeSsrcInfo.forEach(function(lines, idx) {
-        lines = lines.split('\r\n');
-        lines.pop(); // remove empty last element;
-        lines.forEach(function(line) {
-            remoteSdp.media[idx] = remoteSdp.media[idx].replace(line + '\r\n', '');
+    /**
+     * Do a new o/a flow using the existing remote description
+     * @param {SDP object} optionalRemoteSdp optional remote sdp
+     *  to use.  If not provided, the remote sdp from the
+     *  peerconnection will be used
+     * @returns {Promise} promise which resolves when the
+     *  o/a flow is complete with no arguments or
+     *  rejects with an error {string}
+     */
+    _renegotiate (optionalRemoteSdp) {
+        const media_constraints = this.media_constraints;
+        const remoteSdp
+            = optionalRemoteSdp
+                || new SDP(this.peerconnection.remoteDescription.sdp);
+        const remoteDescription = new RTCSessionDescription({
+            type: "offer",
+            sdp: remoteSdp.raw
         });
-    });
-    remoteSdp.raw = remoteSdp.session + remoteSdp.media.join('');
 
-    return remoteSdp;
-};
+        //TODO(brian): in the code below there are 2 chunks of code that relate
+        //  to observing changes in local and remove ufrags.  since they
+        //  just need to read and observe the SDPs, we should create the
+        //  notion of an SDP observer in TraceablePeerConnection that
+        //  gets notified of all SDP changes.  Code like the ufrag
+        //  logic below could listen to that and be separated from
+        //  core flows like this.
+        return new Promise((resolve, reject) => {
+            let remoteUfrag = JingleSessionPC.getUfrag(remoteDescription.sdp);
+            if (remoteUfrag != this.remoteUfrag) {
+                this.remoteUfrag = remoteUfrag;
+
```

Comment:
style: put the body of the function on the next line
style: no parens needed for single parameter arrow functions